fastapi_events/dispatcher.py

```diff
@@ -176,7 +176,82 @@ def dispatch(
     payload_schema_dump: bool = True
 ) -> None:
     """
-    A wrapper of the main dispatcher function with additional checks.
+    Dispatches an event. This is a wrapper of the main dispatcher function with additional checks.
+
+    ### Args
+
+    :param event_name_or_model: Event name or pydantic model. If `event_name` is not provided, \
+        the event name will be derived from the model's `__event_name__` attribute.
+    :param payload: Event payload.
+    :param event_name: Event name. If provided, overrides the event name derived from the event model.
+    :param validate_payload: Validate payload with its registered payload schema. Pydantic payloads \
+        are dumped and re-instantiated.
+    :param payload_schema_cls_dict_args: A dict of args passed to the pydantic dump function. Defaults \
+        to `{"exclude_unset": True}`. See \
+        [pydantic v1](https://docs.pydantic.dev/1.10/usage/exporting_models/#modeldict) or \
+        [pydantic v2](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_dump) \
+        docs for more details.
+    :param payload_schema_registry: Optional payload schema registry. If omitted, the default registry \
+        will be used.
+    :param middleware_id: Optional custom middleware identifier.
+    :param payload_schema_dump: Dump pydantic models to dict before calling event handlers. If False, \
+        the event handlers will be called with instances of the payload schema.
```

Comment:
Hi, first of all, thanks for the awesome work! It's pretty accurate overall. There is just one thing I'm not so sure about, which is the last part of the description of `payload_schema_dump`:
> ... If False,  the event handlers will be called with instances of the payload schema.

I believe the event handlers will be called with any payload passed in, not necessarily instances of payload schema, right? Imagine someone using the dispatch function without pydantic, they could even call it with a custom object and it will still be passed along to the event handlers. Maybe we can refine the last part of the description?

Reply:
Yeah, good catch. How about this:

> Dump pydantic model payloads and validated payloads to dict before calling event handlers. If False, the event handlers will be called with the original payload object.

Reply:
Yeah, good catch. There are interactions between `validate_payload` and the `payload_schema_...` args that make it hard to succinctly describe the behaviours in a docstring.

How about we omit `If False...` and just have this:

> Dump pydantic model payloads and validated payloads to dict before calling event handlers.

Then I'll follow up with detailing the interactions in the README.

Reply:
Sounds good! :+1: